Add support for `init` on services

vdemeester opened the pull request:
It's already supported by `swarmkit`, and act the same as
`HostConfig.Init` on container creation.

<del>Depends on https://github.com/docker/swarmkit/pull/2652</del>
<del>Depends on https://github.com/moby/moby/pull/37059</del>
Required for https://github.com/docker/cli/pull/479

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

thaJeztah commented on vendor/github.com/docker/swarmkit/api/deepcopy/copy.go:
```diff
@@ -44,6 +44,9 @@ func Copy(dst, src interface{}) {
 	case *types.Timestamp:
 		src := src.(*types.Timestamp)
 		*dst = *src
+	case *types.BoolValue:
```
Ah, yes, ran into this one as well :smile:

thaJeztah commented:
Just so that I don't forget (haven't checked yet);

- This option needs to be on `docker service update` as well
- We need a test to verify that running `docker service update` without passing the `--init` option preserves the existing value
- We need a test to verify that the _daemon_ option (which can set `--init` so that it's added by default) is used if the `--init` flag is not passed, on `docker service create`

vdemeester commented:
Updated with a test :angel:

vdemeester committed:
Add support for `init` on services

It's already supported by `swarmkit`, and act the same as
`HostConfig.Init` on container creation.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

thaJeztah reviewed (approved):
LGTM, thanks!

thaJeztah commented:
Janky failing on a flaky test; https://github.com/moby/moby/issues/37132

```
22:35:10 --- FAIL: TestCreateServiceSecretFileMode (32.28s)
22:35:10 	daemon.go:289: [d826463c0a601] waiting for daemon to start
22:35:10 	daemon.go:321: [d826463c0a601] daemon started
22:35:10 	create_test.go:244: timeout hit after 30s: task count at 1 waiting for 0
22:35:10 	daemon.go:279: [d826463c0a601] exiting daemon
```

tiborvass commented on api/swagger.yaml:
```diff
@@ -2721,6 +2721,10 @@ definitions:
               - "process"
               - "hyperv"
+          Init:
+            description: "Run an init inside the container that forwards signals and reaps processes. This field is omitted if empty, and the default (as configured on the daemon) is used."
```
correction: `an init *process*`, can be done in followup

tiborvass reviewed (approved):
LGTM
